src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -0,0 +1,185 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""A data window that moves with the latest datapoints of a data stream."""
+
+import asyncio
+import logging
+from collections.abc import Sequence
+from datetime import datetime, timedelta
+from typing import Union
+
+import numpy as np
+from frequenz.channels import Receiver
+from numpy.typing import ArrayLike
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+log = logging.getLogger(__name__)
+
+
+class MovingWindow(Sequence):
+    """
+    A data window that moves with the latest datapoints of a data stream.
+
+    After initialization the `MovingWindow` can be accessed by an integer
+    index or a timestamp. A sub window can be accessed by using a slice of integers
+    integers or timestamps.
+
+    Note that a numpy ndarray is returned and thus users can use
+    numpys operations directly on a window.
+
+    The window uses an ringbuffer for storage and the first element is aligned to
+    a fixed defined point in time. Since the moving nature of the window, the
+    date of the first and the last element are constantly changing and therefore
+    the point in time that defines the alignment can be outside of the time window.
+    Modulo arithmetic is used to move the `window_alignment` timestamp into the
+    latest window.
+    If for example the `window_alignment` parameter is set to `datetime(1, 1, 1)`
+    and the window size is bigger then one day then the first element will always
+    be aligned to the midnight. For further information see also the
+    `OrderedRingBuffer` documentation.
```

Comment:
```suggestion
    [`OrderedRingBuffer`][frequenz.sdk.timeseries._ringbuffer.OrderedRingBuffer] documentation.
```

It is a bit ugly to add cross-links, but in this case it might be worth it.

PS: We probably need to expose the `OrderedRingBuffer` class publicly (i.e. either make the module `frequenz.sdk.timeseries.ringbuffer` (no leading `_`) or do an import in `frequenz/sdk/timeseries/__init__.py`. I think the former is the best approach, to avoid potential dependency problems and also in case we add more ring buffer implementations.